cpp/log/cert.cc

```diff
@@ -837,6 +887,32 @@ bool IsValidRedactedHost(const string& hostname) {
 
 namespace {
 
+util::Status ExtractSubjectAltNames(STACK_OF(GENERAL_NAME)* subject_alt_names,
+                                    vector<string>* dns_alt_names) {
+  if (subject_alt_names) {
+    const int subject_alt_name_count = sk_GENERAL_NAME_num(subject_alt_names);
+
+    for (int i = 0; i < subject_alt_name_count; ++i) {
+      GENERAL_NAME* const name(sk_GENERAL_NAME_value(subject_alt_names, i));
+
+      util::Status name_status;
+      if (name->type == GEN_DNS) {
+        const string dns_name =
+            ASN1ToStringAndCheckForNulls(name->d.dNSName, "DNS name",
```

Comment:
Wow, thanks for bringing this function to my attention, it needs a bit of love (returning a `util::Status` as an out parameter, backward from the rest?!?). Leave it to me, though. :wink: